Stp.TestingApi/Controllers/TestCategoryController.cs

```diff
@@ -20,41 +21,112 @@ public TestCategoryController(TestingDbContext db)
             _db = db;
         }
 
-        /* https://localhost:5001/api/TestCategory/GetCategories */
         [HttpGet(nameof(GetCategories))]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
         public List<TestCategoryDto> GetCategories()
         {
-            throw new NotImplementedException();
+            var res = _db.TestCategories.Select(x => new TestCategoryDto()
+            {
+                Id = x.Id,
+                Name = x.Name,
+                ParentId = x.ParentId,
+                Position = x.Position,
+                Tests = new List<TestSummaryDto>()
+            }).ToList();
+
+            return res;
         }
 
         /// <returns>Id of just added category</returns>
         [HttpPost(nameof(CreateCategory))]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
         public long CreateCategory(CreateCategoryCommand cmd)
         {
-            throw new NotImplementedException();
+            if (cmd == null)
+            {
+                return -1;
+            }
+
+            var category = new TestCategory()
+            {
+                Name = cmd.Name,
+                ParentId = cmd.ParentCategoryId
+            };
+
+            _db.TestCategories.Add(category);
+            _db.SaveChanges();
+
+            return category.Id;
         }
 
         [HttpPut(nameof(UpdateCategoryName))]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
         public IActionResult UpdateCategoryName(long categoryId, [FromBody]string name)
         {
-            throw new NotImplementedException();
+            var category = _db.TestCategories.Find(categoryId);
+
+            if (category == null)
+            {
+               return NotFound($"Test category with id={categoryId} doesn't exist");
+            }
+
+            category.Name = name;
+            _db.SaveChanges();
+
+            return Ok();
         }
 
         /// <summary>
         /// Changes position of the category in the tree
         /// </summary>
         [HttpPut(nameof(MoveCategory))]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
         public IActionResult MoveCategory(MoveCategoryCommand cmd)
         {
-            throw new NotImplementedException();
+            var categories = _db.TestCategories.Where(c => c.ParentId == cmd.ParentCategoryId).ToList().OrderBy(c => c.Position);
```

Comment:
Need to check whether the parent category exists.
If the parent category contains less than 2 nodes - throw BadRequest.